fedimint-client/src/lib.rs

```diff
@@ -711,7 +717,6 @@ struct ClientInner {
     db: Database,
     federation_id: FederationId,
     federation_meta: BTreeMap<String, String>,
-    primary_module: DynPrimaryClientModule,
     primary_module_instance: ModuleInstanceId,
     primary_module_kind: ModuleKind,
```

Comment:
The primary module kind isn't needed anymore. The only reason we kept it around was because the kind wasn't saved in the module registry like for all other modules:

https://github.com/fedimint/fedimint/blob/cc179f060e1029139a1eab69f98033be7189b14e/fedimint-client/src/lib.rs#L631-L641